metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -653,7 +653,7 @@ def _build_plan_for_distinct_values(self, query_spec: MetricFlowQuerySpec) -> Da
             linkable_spec_set=required_linkable_specs, time_range_constraint=query_spec.time_range_constraint
         )
         if not dataflow_recipe:
-            raise UnableToSatisfyQueryError(f"Recipe not found for linkable specs: {required_linkable_specs}")
+            raise UnableToSatisfyQueryError(f"Unable to join all items in request: {required_linkable_specs}")
```

Comment:
Is this user-facing, and do we have any idea know when it happens and why? If so we could maybe add some more details around possible causes, but at this point I don't think we have that info yet.

I'm kind of wondering why this happens.

Reply:
Yeah, it's a good question! I did a quick DD search of the past week of these errors: https://dbtlabsmt.datadoghq.com/logs?query=recipe%20not%20found%20&cols=host%2Cservice&fromUser=true&index=%2A&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1712081485492&to_ts=1712686285492&live=true
There is some stuff I don't understand here, like how a non-additive dimension spec might impact the dataflow plan.